filters/builtin/headerfilter.go

```diff
@@ -281,7 +283,12 @@ func (f *headerFilter) Request(ctx filters.FilterContext) {
 			ctx.SetOutgoingHost(f.value)
 		}
 	case dropRequestHeader:
-		header.Del(f.key)
+		if f.value == "" {
+			header.Del(f.key)
+		} else {
+			k := textproto.CanonicalMIMEHeaderKey(f.key)
+			header[k] = slices.DeleteFunc(header[k], func(v string) bool { return v == f.value })
```

Comment:
I think value should also be case insensitive.

Reply:
My idea was to have a low-level primitive to drop a specific value. 
For setting and adding header we do not do any modification to the value.

To support regexp we may consider changing modRequestHeader such that it _deletes_ the matching header if "the replacement (string)" parameter is empty (or absent) or maybe add a whole new filter.

Reply:
I think we should add another filter dropRequestHeaderRegexp maybe?

Reply:
Could a third boolean parameter to enable case-insensitive comparison be an alternative to regex?

Reply:
It could be done like that of course but lets wait for the usecase.

I initially considered to add a new filter `dropRequestHeaderValue` then decided to extend existing one but we should wary about how much logic we cram into it.

These header filters are basic primitives and building blocks and I think we should keep them simple. For higher-level or interesting logic we can always add new filters.

Reply:
> I think we should add another filter dropRequestHeaderRegexp maybe?

Indeed, thanks, a new dropRequestHeaderRegexp would be better because exact value match is a very simple regexp (`foo -> ^foo$` ), stdlib handles it as a special case of fast prefix check, we already use it e.g. in HeaderRegexp and it covers many other usecases.

Reply:
> Could a third boolean parameter to enable case-insensitive comparison be an alternative to regex?

This could be achieved by regexp flags (e.g. `^(?i)foo$`)